addons/im_livechat/models/discuss_channel.py

```diff
@@ -511,19 +555,28 @@ def _message_post_after_hook(self, message, msg_vals):
                 }
             )
 
-        # sudo: discuss.channel - accessing history for checking if author is an agent is acceptable
-        author_is_agent = bool(
-            self.sudo().livechat_channel_member_history_ids.filtered(
-                lambda h: h.partner_id == message.author_id and h.livechat_member_type == "agent",
-            ),
-        )
-        if not self.livechat_end_dt and author_is_agent:
+        author_history = self.env["im_livechat.channel.member.history"]
+        # sudo - discuss.channel: accessing history to update its state is acceptable
+        if message.author_id or message.author_guest_id:
+            author_history = self.sudo().livechat_channel_member_history_ids.filtered(
+                lambda h: h.partner_id == message.author_id
+                if message.author_id
+                else h.guest_id == message.author_guest_id
+            )
```

Comment:
I believe there is still a potential issue here. 
If the message doesn't have any author_id and guest_id, it would incorrectly match any history where guest_id is false.
Not a realistic flow in channels but still technically possible.